### PR TITLE
remove EnableNonGCPMode flag from NEG controller

### PR DESCRIPTION
**Pull request by freehan**
I will push separate PRs for different flags.

### CODE REVIEW

File: pkg/neg/manager.go [hunk 1]
```diff
@@ -87,23 +86,38 @@ type syncerManager struct {
 
 	// kubeSystemUID is used to by syncers when NEG CRD is enabled
 	kubeSystemUID types.UID
+
+	// indicates whether nonGcpMode have been enabled
```
Comment (rramkumar1) [inline pkg/neg/manager.go:90]: would add a comment too indicating what this actually implies in terms of how the controller operates

**Review by rramkumar1 — COMMENTED** [review]
One nit

File: pkg/neg/manager.go (hunk 1 above)
Comment (freehan) [reply pkg/neg/manager.go]: Done. Fixed

**Pull request merged**